File: ClientApplication/scip-client-application/backend/src/main/java/blockchains/iaas/uni/stuttgart/de/scipcasestudy/clientapplication/backend/model/response/rest/SeafoodOccurrence.java
package blockchains.iaas.uni.stuttgart.de.scipcasestudy.clientapplication.backend.model.response.rest;

import blockchains.iaas.uni.stuttgart.de.scipcasestudy.clientapplication.backend.model.common.rest.SupplyChainAction;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class SeafoodOccurrence<T extends SupplyChainAction> {
    private String isoTimestamp;
    private T occurrence;
}

File: ClientApplication/scip-client-application/frontend/src/app/models/SeafoodOccurrence.ts
import Fish from './Fish';
import FishPackage from './FishPackage';
import InventoryEntry from './InventoryEntry';
import PackageSelling from './PackageSelling';
import FishShipment from './FishShipment';
import PackageTransportation from './PackageTransportation';
import SupplyChainAction from './SupplyChainAction';

export default interface SeafoodOccurrence <T extends SupplyChainAction> {
    isoTimestamp: string;
    occurrence: T;
}
